File: Easy/2026-04-15-2515-Shortest-Distance-to-Target-String-in-a-Circular-Array/solution.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        if target not in words:
            return -1
        
        left_distance = float('inf')
        right_distance = float('inf')
        
        for i in range(n):
            if words[i] == target:
                left_distance = min(left_distance, (startIndex - i) % n)
                right_distance = min(right_distance, (i - startIndex) % n)
        
        return min(left_distance, right_distance)

File: Easy/2026-04-15-2515-Shortest-Distance-to-Target-String-in-a-Circular-Array/solution.js
var closetTarget = function(words, target, startIndex) {
    const n = words.length;
    let minDistance = Infinity;

    for (let i = 0; i < n; i++) {
        if (words[i] === target) {
            const distance = Math.min(Math.abs(i - startIndex), n - Math.abs(i - startIndex));
            minDistance = Math.min(minDistance, distance);
        }
    }

    return minDistance === Infinity ? -1 : minDistance;
};